auth/src/main/java/com/firebase/ui/auth/ui/accountmanagement/AccountSettingsActivity.java

```diff
@@ -0,0 +1,269 @@
+        });
+
+        editEmail.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivityForResult(getReauthIntent(), RC_REAUTH_EDIT_EMAIL);
+            }
+        });
+
+        Button changePassword = (Button) findViewById(R.id.change_password);
+        changePassword.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                startActivityForResult(getReauthIntent(), RC_REAUTH_PASSWORD);
+            }
+        });
+
+        populateView();
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        switch (requestCode) {
+            case RC_EDIT_DISPLAY_NAME:
+            case RC_EDIT_EMAIL:
+                populateView();
+                break;
+            case RC_REAUTH_EDIT_EMAIL:
+                switch (resultCode) {
+                    case RESULT_OK:
+                        startActivityForResult(
+                                EditEmailActivity.createIntent(
+                                        getApplicationContext(),
+                                        mFlowParameters),
+                                RC_EDIT_EMAIL);
+                        break;
+                    default:
+                        // fall through
+                }
+                break;
+            case RC_REAUTH_PASSWORD:
+                switch (resultCode) {
+                    case RESULT_OK:
+                        startActivityForResult(
+                                EditPasswordActivity.createIntent(this, mFlowParameters),
+                                RC_CHANGE_PASSWORD);
+                        break;
+                    default:
+                        // fall through
+                }
+                break;
+            case RC_CHANGE_PASSWORD:
+            default:
+                // fall through
+        }
+    }
+
+    private void populateView() {
+        FirebaseUser firebaseUser = mActivityHelper.getCurrentUser();
+        mDisplayName.setText(firebaseUser.getDisplayName());
+        mEmail.setText(firebaseUser.getEmail());
+        populateLinkedAccounts();
+        maybeShowChangePassword();
+    }
+
+    private void populateLinkedAccounts() {
+        View linkedAccountsSection = findViewById(R.id.linked_accounts_section);
+        FirebaseUser user = mActivityHelper.getCurrentUser();
+        if (user == null) {
+            return;
+        }
+        List<String> providers = user.getProviders();
+        if (providers == null || providers.isEmpty()) {
+            linkedAccountsSection.setVisibility(View.GONE);
+            return;
+        } else {
+            linkedAccountsSection.setVisibility(View.VISIBLE);
+        }
+        ViewGroup linkedAccountHolder = (ViewGroup) findViewById(R.id.linked_accounts_holder);
```

Comment:
This whole section below is a huge amount of display logic to have here.  Could we refactor into a custom ProviderView that takes a providerId?